tensorflow/core/kernels/maxpooling_op_gpu.cu.cc

```diff
@@ -76,7 +91,10 @@ __global__ void MaxPoolForwardNCHW(const int nthreads, const dtype* bottom_data,
     for (int h = hstart; h < hend; ++h) {
       for (int w = wstart; w < wend; ++w) {
         int idx = c * height * width + h * width + w;
-        if (bottom_data_n[idx] > maxval) {
+        // Note: The first inequality below ensures NaNs are propagated.
+        // Template argument used to avoid conditional eval in inner loop.
+        if (   (  propagate_nans && !(bottom_data_n[idx] <= maxval) )
```

Comment:
Also please run the benchmarks and put in performance comparison before/after your change in the commit description. Thanks!

Reply:
Is there a targeted benchmark for this operation that you have in mind?

Reply:
How about the benchmarks in tensorflow/core/kernels/nn_ops_test.cc?